File: app/src/main/java/com/popalay/yocard/injection/AppModule.java
package com.popalay.yocard.injection;

import android.content.Context;

import com.popalay.yocard.App;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final Context mApplicationContext;

    public AppModule(App app) {
        mApplicationContext = app;
    }

    @Provides
    @Singleton
    Context provideApplicationContext() {
        return mApplicationContext;
    }

}

File: presentation/src/main/kotlin/com/popalay/cardme/injection/AppModule.kt
package com.popalay.cardme.injection

import android.content.Context
import com.popalay.cardme.App
import com.popalay.cardme.base.navigation.CustomRouter
import dagger.Module
import dagger.Provides
import ru.terrakok.cicerone.Cicerone
import ru.terrakok.cicerone.NavigatorHolder
import javax.inject.Singleton

@Module
class AppModule {

    private val cicerone = Cicerone.create(CustomRouter())

    @Provides
    fun provideApplicationContext(app: App): Context = app.applicationContext

    @Provides @Singleton
    fun provideRouter(): CustomRouter = cicerone.router

    @Provides @Singleton
    fun provideNavigatorHolder(): NavigatorHolder = cicerone.navigatorHolder

}